storage/src/test/java/tech/pegasys/teku/storage/server/kvstore/DatabaseTest.java

```diff
@@ -1926,16 +1994,36 @@ private UpdatableStore recreateStore() {
   }
 
   private void createStorageSystemInternal(
```

Comment:
I'd be very tempted to essentially move this method into `DatabaseContext`. You'd have to pass in the tmpDir and `spec` I think but that's easy enough and not an issue to create a tmp dir even if it doesn't wind up being needed. Would just make the `DatabaseContext` more of a standalone provider rather than just a thin wrapper around a couple of config values.